cadc-util/src/main/java/ca/nrc/cadc/auth/ServletPrincipalExtractor.java

```diff
@@ -117,10 +123,34 @@ private ServletPrincipalExtractor()
     public ServletPrincipalExtractor(final HttpServletRequest req)
     {
         this.request = req;
+        
+        // support certs from the java request attribute and from
+        // the request header.  favour the request attribute for now.
         X509Certificate[] ca = (X509Certificate[])
             request.getAttribute(CERT_REQUEST_ATTRIBUTE);
-
-        if (!ArrayUtil.isEmpty(ca)) {
+        
+        if (ArrayUtil.isEmpty(ca)) {
+            // try the header field
+            String certString = req.getHeader(CERT_HEADER_FIELD);
+            log.debug("Cert String: " + certString);
+            if (certString != null && certString.length() > 0) {
+                try {
+                    byte[] certBytes = Base64.decode(certString);
+                    if (certBytes != null) {
```

Comment:
This code should just use SSLUtil.getCertifcates(byte[]) which does Base64 decoding and then SSLUtil.readCertificateChain(byte[]) which does the rest.